From 756c306502498f999fdd494477b9cea1b45e4faf Mon Sep 17 00:00:00 2001
From: Stefan Liebler <stli@linux.ibm.com>
Date: Fri, 21 Aug 2020 11:23:17 +0200
Subject: [PATCH] S390: Sync HWCAP names with kernel by adding aliases [BZ
 #25971]

Unfortunately some HWCAP names like HWCAP_S390_VX differs between
kernel (see <kernel>/arch/s390/include/asm/elf.h) and glibc.

Therefore, those HWCAP names from kernel are now introduced as alias
---
 sysdeps/s390/dl-procinfo.h                | 3 +++
 sysdeps/unix/sysv/linux/s390/bits/hwcap.h | 3 +++
 2 files changed, 6 insertions(+)

diff --git a/sysdeps/s390/dl-procinfo.h b/sysdeps/s390/dl-procinfo.h
index 0db4bc39c7..08eee109f7 100644
--- a/sysdeps/s390/dl-procinfo.h
+++ b/sysdeps/s390/dl-procinfo.h
@@ -51,8 +51,11 @@ enum
   HWCAP_S390_HIGH_GPRS = 1 << 9,
   HWCAP_S390_TE = 1 << 10,
   HWCAP_S390_VX = 1 << 11,
+  HWCAP_S390_VXRS = HWCAP_S390_VX,
   HWCAP_S390_VXD = 1 << 12,
+  HWCAP_S390_VXRS_BCD = HWCAP_S390_VXD,
   HWCAP_S390_VXE = 1 << 13,
+  HWCAP_S390_VXRS_EXT = HWCAP_S390_VXE,
   HWCAP_S390_GS = 1 << 14,
   HWCAP_S390_VXRS_EXT2 = 1 << 15,
   HWCAP_S390_VXRS_PDE = 1 << 16,
diff --git a/sysdeps/unix/sysv/linux/s390/bits/hwcap.h b/sysdeps/unix/sysv/linux/s390/bits/hwcap.h
index 6adbec018b..f2998ff131 100644
--- a/sysdeps/unix/sysv/linux/s390/bits/hwcap.h
+++ b/sysdeps/unix/sysv/linux/s390/bits/hwcap.h
@@ -36,8 +36,11 @@
 #define HWCAP_S390_HIGH_GPRS    512
 #define HWCAP_S390_TE           1024
 #define HWCAP_S390_VX           2048
+#define HWCAP_S390_VXRS         HWCAP_S390_VX
 #define HWCAP_S390_VXD          4096
+#define HWCAP_S390_VXRS_BCD     HWCAP_S390_VXD
 #define HWCAP_S390_VXE          8192
+#define HWCAP_S390_VXRS_EXT     HWCAP_S390_VXE
 #define HWCAP_S390_GS           16384
 #define HWCAP_S390_VXRS_EXT2    32768
 #define HWCAP_S390_VXRS_PDE     65536
-- 
2.31.1

